### ISSUE TITLE
A discussion towards better understanding of the 1st task

### ISSUE THREAD

**Issue by JoeHEZHAO** [vgsatorras/hybrid-inference#3]
Dear Authors

Thanks for your great work. I am a big fun, but I have few questions towards the big picture of your work.

May I ask:  Does the task of exp1_linear.py aim at recovering true trajectory positions from noisy measurements ? I notice the loss is calculated from MSE between positions and meas as in 
 `outputs = net([operators, meas], x0, args.K, ts=ts)`
`mse = F.mse_loss(outputs[-1], position)`

If so, Does this means that, for hybrid model, we introduce Graphical Model Message (GMM) and Graph Neural Network (GNN) mechanism to better model the unobserved variable x, for the purpose that x can flexibly take non-gaussian noise influence into account ?

Thanks for your time !!

**Comment by vgsatorras** [comment]
Hi Joe, 

Thank you for the interest in our work. 

> Does the task of exp1_linear.py aim at recovering true trajectory positions from noisy measurements? 

Yes, it does.

The aim of our model is to enhance the Graphical Model Messages (GMM) estimation using a Graph Neural Network. In many real world scenarios a graphical model is an approximation of the true data generating process and that leads to suboptimal estimates. In our model, the GNN improves the GMM performance by learning these complexities from data. These complexities may take any non-analytical form, as you mentioned they may also be non-gaussian. 

Experiment 1 is a proof of concept where we generate some trajectories and we use a simpler graphical model for the GMM messages. Then we enhance the GMM messages with a GNN that is trained from data.

Best,
Víctor

**Comment by JoeHEZHAO** [comment]
Victor

Thanks for your timely reply. You explaination make a lot of sense to me. I closed this thread since this discussion is not about code issue. 

It is interesting to see that a hybrid model can outperform both the pure equation-based and pure learning-based methods. Thanks again !
